tabml/data_processing.py

```diff
@@ -68,12 +68,12 @@ def __init__(self, whisker_coeff: float = 1.5):
         self.lower: float = float("-inf")
         self.upper: float = float("inf")
 
-    def fit(self, X: pd.Series):
+    def fit(self, X: pd.DataFrame, y=None):
         self.lower, self.upper = find_boxplot_boundaries(X, self.whisker)
         return self
 
-    def transform(self, X):
-        return X.clip(self.lower, self.upper, axis=0)
+    def transform(self, X: pd.DataFrame):
+        return X.clip(self.lower, self.upper, axis="columns")
```

Comment:
`axis=0` will take the clip row-wise (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.quantile.html).